Mathlib/Data/FP/Basic.lean

```diff
@@ -0,0 +1,301 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+
+! This file was ported from Lean 3 source module data.fp.basic
+! leanprover-community/mathlib commit 7b78d1776212a91ecc94cf601f83bdcc46b04213
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Semiquot
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Implementation of floating-point numbers (experimental).
+-/
+
+-- Porting note: TODO add docs and remove `@[nolint docBlame]`
+
+@[nolint docBlame]
+def Int.shift2 (a b : ℕ) : ℤ → ℕ × ℕ
+  | Int.ofNat e => (a.shiftl e, b)
+  | Int.negSucc e => (a, b.shiftl e.succ)
+#align int.shift2 Int.shift2
+
+namespace FP
+
+@[nolint docBlame]
+inductive RMode
+  | NE -- round to nearest even
+  deriving Inhabited
+#align fp.rmode FP.RMode
+
+@[nolint docBlame]
+class FloatCfg where
+  (prec emax : ℕ)
+  prec_pos : 0 < prec
+  prec_max : prec ≤ emax
+attribute [nolint docBlame] FloatCfg.prec FloatCfg.emax FloatCfg.prec_pos FloatCfg.prec_max
+#align fp.float_cfg FP.FloatCfg
+
+variable [C : FloatCfg]
+
+-- Porting note: no longer needed
+-- include C
+
```

Comment:
```suggestion
```
This is not something that we'll need to return to later, so it's fine to just remove it.